packages/lib/src/vis/shared/SelectionMesh.tsx

```diff
@@ -35,25 +56,31 @@ function SelectionMesh(props: Props) {
       }
 
       (target as Element).setPointerCapture(pointerId);
-      const point = worldToData(unprojectedPoint);
+      const point = project(unprojectedPoint);
 
       setStartPoint(point);
       setEndPoint(point);
       setDrag(true);
+      if (onSelectionStart) {
+        onSelectionStart();
+      }
     },
-    [worldToData, modifierKey]
+    [modifierKey, onSelectionStart, project]
   );
 
   const onPointerMove = useCallback(
     (evt: ThreeEvent<PointerEvent>) => {
       if (!isDragging) {
         return;
       }
-      const point = worldToData(evt.unprojectedPoint);
+      const point = project(evt.unprojectedPoint);
 
       setEndPoint(point);
+      if (startPoint && onSelectionChange) {
```

Comment:
I wonder if you still need `isDragging` now? You may be able to just rely on `startPoint` being defined (in which case the early return would allow removing it from this condition).

Reply:
Excellent analysis :+1: